src/circuits/validator.ts

```diff
@@ -0,0 +1,124 @@
+};
+const credentialAtomicQueryV2OnChainValidation = {
+  validation: { maxQueriesCount: 1, supportedOperations: v2OnChainOperations }
+};
+const credentialAtomicQueryV3Validation = {
+  validation: { maxQueriesCount: 1, supportedOperations: allOperations }
+};
+
+export const circuitValidator: {
+  [k in CircuitId]: {
+    validation: { maxQueriesCount: number; supportedOperations: Operators[] };
+    subVersions?: {
+      mtLevel?: number;
+      mtLevelClaim?: number;
+      mtLevelOnChain?: number;
+      queryCount?: number;
+      targetCircuitId: CircuitId;
+    }[];
+  };
+} = {
+  [CircuitId.AtomicQueryMTPV2]: credentialAtomicQueryV2Validation,
+  [CircuitId.AtomicQueryMTPV2OnChain]: credentialAtomicQueryV2OnChainValidation,
+  [CircuitId.AtomicQuerySigV2]: credentialAtomicQueryV2Validation,
+  [CircuitId.AtomicQuerySigV2OnChain]: credentialAtomicQueryV2OnChainValidation,
+  [CircuitId.AtomicQueryV3]: credentialAtomicQueryV3Validation,
+  [CircuitId.AtomicQueryV3OnChain]: credentialAtomicQueryV3Validation,
+  [CircuitId.AuthV2]: noQueriesValidation,
+  [CircuitId.AuthV3]: {
+    ...noQueriesValidation,
+    subVersions: [
+      {
+        targetCircuitId: (CircuitId.AuthV3 + '-8-32') as CircuitId,
+        mtLevel: 8,
+        mtLevelOnChain: 32
+      }
+    ]
+  },
+  [CircuitId.StateTransition]: noQueriesValidation,
+  [CircuitId.LinkedMultiQuery10]: {
+    validation: { maxQueriesCount: 10, supportedOperations: allOperations }
+  },
+
+  [CircuitId.AtomicQueryV3Stable]: {
+    ...credentialAtomicQueryV3Validation,
+    subVersions: [
+      {
+        mtLevel: 16,
+        mtLevelClaim: 16,
+        targetCircuitId: (CircuitId.AtomicQueryV3Stable + '-16-16-64') as CircuitId
+      }
+    ]
+  },
+  [CircuitId.AtomicQueryV3OnChainStable]: {
+    ...credentialAtomicQueryV3Validation,
+    subVersions: [
+      {
+        mtLevel: 16,
+        mtLevelClaim: 16,
+        mtLevelOnChain: 32,
+        targetCircuitId: (CircuitId.AtomicQueryV3OnChainStable + '-16-16-64-16-32') as CircuitId
+      }
+    ]
+  },
+  [CircuitId.LinkedMultiQueryStable]: {
+    validation: { maxQueriesCount: 10, supportedOperations: allOperations },
+    subVersions: [
+      {
+        queryCount: 3,
+        targetCircuitId: (CircuitId.LinkedMultiQueryStable + '3') as CircuitId
+      },
+      {
+        queryCount: 5,
+        targetCircuitId: (CircuitId.LinkedMultiQueryStable + '5') as CircuitId
+      },
+      {
+        queryCount: 10,
+        targetCircuitId: (CircuitId.LinkedMultiQueryStable + '10') as CircuitId
+      }
+    ]
+  }
+};
+
+export const getCircuitIdsWithSubVersions = (filterCircuitIds?: CircuitId[]): CircuitId[] => {
+  return Object.keys(circuitValidator).reduce((acc, key) => {
+    const circuitId = key as CircuitId;
+    // if filterCircuitIds is provided, only include circuits that are in the filterCircuitIds, else include all circuits
+    if (filterCircuitIds && !filterCircuitIds.includes(circuitId)) {
```

Comment:
if I pass linkedMultiQuery 

result will be: 
linkedMultiQuery10
linkedMultiQuery5
linkedMultiQuery3
linkedMultiQuery10-beta.1

but I should recieve only 

linkedMultiQuery10
linkedMultiQuery5
linkedMultiQuery3


if I pass credentialAtomicQueryV3
result should be: 
credentialAtomicQueryV3-16-16-64
credentialAtomicQueryV3

please confirm